src/handler.ts

```diff
@@ -1,14 +1,115 @@
+  });
 
-const handler = async (): void => {
-  logger.debug("Function triggered'.");
+  const results = await Promise.allSettled(altPromiseArray);
+  const ids = results
+    .map((result, index) => (result.status === 'fulfilled' ? null : event.Records[index].messageId))
+    .filter((item) => item !== null);
+  return {
+    batchItemFailures: ids.map((id) => ({ itemIdentifier: id })),
+  };
+};
+
+const generateAndUpload = async (documentData, request: Request, fileName: string) => {
+  try {
+    logger.info('Starting lambda to lambda invoke');
+    const response = await invokePdfGenLambda(documentData, request.documentName);
+    logger.info('Finished lambda to lambda invoke, checking response');
+
+    const responseString: string = new TextDecoder().decode(response.Payload);
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const responseJson: any = JSON.parse(responseString);
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    const responseBuffer: Buffer = Buffer.from(responseJson.body, 'base64');
+
+    const metaData = {
+      'date-of-issue': Date.now().toString(),
+      'cert-type': request.documentName,
+      'file-format': 'pdf',
+      'file-size': responseBuffer.byteLength.toString(),
+      'should-email-certificate': 'false',
+    };
+    logger.info(`Starting s3 upload for file: ${process.env.BRANCH}/${fileName}`);
+    await uploadPdfToS3(responseBuffer, metaData, fileName);
+    logger.info('Finished s3 upload');
+  } catch (error) {
+    logger.error(error);
+    throw error;
+  }
 };
 
-export { handler };
+const invokePdfGenLambda = async (docGenPayload, documentType: string, lambdaClient?: LambdaClient) => {
+  const client = lambdaClient ?? new LambdaClient({ region: process.env.AWS_REGION });
+  const payload: any = JSON.stringify({
+    httpMethod: 'POST',
+    pathParameters: {
+      documentName: documentType,
+      documentDirectory: 'CVS',
+    },
+    json: true,
+    body: JSON.stringify(docGenPayload),
+  });
+  const command = new InvokeCommand({
+    FunctionName: process.env.DOC_GEN_NAME,
+    InvocationType: 'RequestResponse',
+    LogType: 'Tail',
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    Payload: payload,
+  });
+
+  const response = await client.send(command);
+  return response;
+};
+
+const uploadPdfToS3 = async (
+  data: Buffer | Uint8Array | Blob | string | Readable,
+  metadata: Record<string, string>,
+  fileName: string,
+  s3?: S3,
+): Promise<any> => {
+  const s3Client = s3 ?? new S3({ region: 'eu-west-1' });
+  return s3Client
+    .send(
+      new PutObjectCommand({
+        Bucket: process.env.BUCKET_NAME,
+        Key: `${process.env.BRANCH}/${fileName}.pdf`,
+        Body: data,
+        Metadata: metadata,
+      }),
+    )
+    .catch((err) => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+      logger.error(err.message);
```

Comment:
This catch needs to re-throw to ensure it can actually fail.

Reply:
Done now :)